Assets/Scripts/Effects/GenerateWaterPlane.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+namespace Effects
+{
+    [ExecuteInEditMode]
+    public class GenerateWaterPlane : MonoBehaviour
+    {
+        [SerializeField][Tooltip("Size of the mesh")]
+        private float size = 1f;
+    
+        [SerializeField][Tooltip("Subdivisions of the mesh")]
+        private int gridSize = 16;
+
+        private MeshFilter _filter;
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            _filter = GetComponent<MeshFilter>();
+            _filter.mesh = generateMesh();
+        }
+
+        private Mesh generateMesh()
+        {
+            Mesh m = new Mesh();
+            List<Vector3> vertices = new List<Vector3>();
+            List<Vector3> normals = new List<Vector3>();
+            List<Vector2> uvs = new List<Vector2>();
+        
+            for(int x = 0; x < gridSize + 1; x++)
+            {
+                for (int y = 0; y < gridSize + 1; y++)
+                {
+                    vertices.Add(new Vector3(-size * .5f + size * (x / ((float)gridSize)), 0, -size * .5f + size * (y / ((float)gridSize))));
+                    normals.Add(Vector3.up);
+                    uvs.Add(new Vector2(x / (float)gridSize, y / (float)gridSize));
+                }
+            }
+
+            List<int> triangles = new List<int>();
+            int vertCount = gridSize + 1;
+
+            for (int i = 0; i < vertCount * vertCount - vertCount; i++)
```

Comment:
I'm not really sure what this does, but shouldn't it be vertices.Size - vertCount - 1 (which is equal to vertCount * vertCount - vertCount - 1) because you're starting at 0?
Why do you subtract vertCount? Don't you miss the last row of vertices then? (because you do skip the iterations of the loop where (i+1)%vertCount == 0 below)

Reply:
it creates a list of vertices to create triangles from.
The last iteration i gets skipped by the modulo so "-1" is not necessary

Reply:
Assume vertCount = 3 --> Max Index: 3*3-3=6 
i = 5 --> 5+1 % 6 == 0 (gets skipped)
i = 6 --> 6+1 % 6 != 0 (not skipped)

Reply:
you're right 😅